src/util/batchInsert.js

```diff
@@ -35,21 +35,23 @@ export default function batchInsert(client, tableName, batch, chunkSize = 1000)
       .then((tr) => {
         return Promise.mapSeries(chunks, (items) => tr(tableName).insert(items, returning))
           .then((result) => {
-            result = flatten(result);
+            result = flatten(result || []);
 
             if(autoTransaction) {
-              return tr.commit()
+              //TODO: -- Oracle tr.commit() does not return a 'thenable' !? Ugly hack for now.
+              return (tr.commit(result) || Promise.resolve())

```

Comment:
I'm planning to setup docker container with postgres, mysql, oracle and mssql maybe with that it will be easier to test locally all dialects (hopefully I have time to do it before 2018 :p and permission from microsoft to bundle mssql there...).